exec/cnex/lib/time_posix.c

```diff
@@ -0,0 +1,40 @@
+#define _POSIX_C_SOURCE 200809L
+#include <stdlib.h>
+#include <sys/time.h>
+#include <time.h>
+#include <unistd.h>
+
+#include "cell.h"
+#include "exec.h"
+#include "number.h"
+#include "stack.h"
+
+extern Number NANOSECONDS_PER_SECOND;
+
+void time_sleep(TExecutor *exec)
+{
+    // Note: The number on the top of the stack is in SECONDS!
+    uint64_t ns = number_to_uint64(number_multiply(top(exec->stack)->number, NANOSECONDS_PER_SECOND)); pop(exec->stack);
+
+    // So we must get the correct NANOSECOND value to sleep the correct amount of time.
+    struct timespec ts;
+    if (ns > number_to_uint64(NANOSECONDS_PER_SECOND)) {
+        ts.tv_sec = (int)(ns / number_to_uint64(NANOSECONDS_PER_SECOND));
+        ts.tv_nsec = (ns - ((long)ts.tv_sec * number_to_uint64(NANOSECONDS_PER_SECOND)));
+   } else {
+        ts.tv_sec = 0;
```

Comment:
I don't see why this has to be an if/else situation. The calculation above is fine, this else clause is unnecessary.

Reply:
Removed unnecessary if/else clause.